app/jobs/slack_event/admins_job.rb

```diff
@@ -0,0 +1,70 @@
+#
+# This comes through the slash commands API which has a
+# slightly different payload than the events API
+#
+# Example payload:
+#
+# {
+#   "token": "xxxx",
+#   "team_id": "TCF0ZD3LL",
+#   "team_domain": "railslink-dev",
+#   "channel_id": "xxxx",
+#   "channel_name": "xxxx",
+#   "user_id": "xxxx",
+#   "user_name": "xxxx",
+#   "command": "/admins",
+#   "text": "",
+#   "response_url": "https://hooks.slack.com/commands/TCF0ZD3LL/640895425681/GDK2vHvOpxpvFrpRJnrvxiHS",
+#   "trigger_id": "xxxx",
+#   "slack": {
+#     "event": {
+#       "type": "admins"
+#     }
+#   },
+#   "controller": "slack",
+#   "action": "event"
+# }
+#
+
+class SlackEvent::AdminsJob < ApplicationJob
+
+  attr_reader :options
+  private :options
+
+  def perform(options = {})
+    @options = options.with_indifferent_access
+    message = {
+      text: admin_text,
+      response_type: "in_channel"
+    }
+    HTTParty.post(URI::parse(@options[:response_url]), body: message.to_json)
+  end
+
+  private
+
+  # Return the text message for the list of admins.
+  #
+  # The text message should not be wrapped as that will break Slack's automatic
+  # wrapping on presentation.
+  #
+  # Returns String (or nil)
+  def admin_text
+  <<~END_OF_TEXT
+    Admins and their contact information:
+    #{admin_string}
+    END_OF_TEXT
+  end
+
+  def admin_string
+    admins_string = ""
+    SlackUser.admins.shuffle.each do |admin|
+      name = [admin.try(:first_name), admin.try(:last_name)].compact.join(" ")
```

Comment:
I'm not sure how much of an toy project this is for you or if you just want the feature knocked out, but this piece might be a decent place to do some refactoring practice with some sort of presenter (or even class methods on the slack user class) to take the logic out of figuring out how to display the name,email,tz,etc from this method, and be re-useable elsewhere
And instead you could do
```ruby
map do |admin|
  [admin.name, admin.email, admin.tz, admin.activity]
    .compact
    .join(" | ")
end,join("\n")
```
(There would be 1 less trailing \n in this way, which might be favorable so you can easier control it in the output message with the join refactor, otherwise can add back this string interpolation way with same concept)

Reply:
Honestly this has been sitting for several months and I'd kind of like to get a quick win, but I'm perfectly happy for us to write some tech debt tickets about refactoring presentation.  At this point, I don't want to hold up the feature for it if others are okay with that.

Reply:
Yeah, this wasn't a "need", it was more of just seeing possible improvements.